modules/javafx.graphics/src/main/java/com/sun/javafx/sg/prism/NGShape3D.java

```diff
@@ -179,61 +171,95 @@ private void renderMeshView(Graphics g) {
                  * intensity *= attenuationFactor;
                 */
-                if (lightBase instanceof NGPointLight) {
-                    var light = (NGPointLight) lightBase;
-                    Affine3D lightWT = light.getWorldTransform();
-                    meshView.setLight(lightIndex++,
-                            (float) lightWT.getMxt(),
-                            (float) lightWT.getMyt(),
-                            (float) lightWT.getMzt(),
-                            rL, gL, bL, 1.0f,
-                            light.getCa(),
-                            light.getLa(),
-                            light.getQa(),
-                            1,
-                            light.getMaxRange(),
-                            (float) light.getDirection().getX(),
-                            (float) light.getDirection().getY(),
-                            (float) light.getDirection().getZ(),
-                            light.getInnerAngle(),
-                            light.getOuterAngle(),
-                            light.getFalloff());
-                } else if (lightBase instanceof NGDirectionalLight) {
-                    var light = (NGDirectionalLight) lightBase;
-                    meshView.setLight(lightIndex++,
-                            0, 0, 0,                 // position is irrelevant
-                            rL, gL, bL, 1.0f,
-                            1, 0, 0,                 // attenuation is irrelevant
-                            0,
-                            Float.POSITIVE_INFINITY, // range is irrelevant
-                            (float) light.getDirection().getX(),
-                            (float) light.getDirection().getY(),
-                            (float) light.getDirection().getZ(),
-                            0, 0, 0);                // spotlight factors are irrelevant
-                } else if (lightBase instanceof NGAmbientLight) {
-                    // Accumulate ambient lights
-                    ambientRed   += rL;
-                    ambientGreen += gL;
-                    ambientBlue  += bL;
+                else if (lightBase instanceof NGSpotLight light) {
+                    addSpotLight(light, lightIndex++, rL, gL, bL);
+                } else if (lightBase instanceof NGPointLight light) {
+                    addPointLight(light, lightIndex++, rL, gL, bL);
+                } else if (lightBase instanceof NGDirectionalLight light) {
+                    addDirectionalLight(light, lightIndex++, rL, gL, bL);
                 }
             }
-            ambientRed = Utils.clamp(0, ambientRed, 1);
-            ambientGreen = Utils.clamp(0, ambientGreen, 1);
-            ambientBlue = Utils.clamp(0, ambientBlue, 1);
-            meshView.setAmbientLight(ambientRed, ambientGreen, ambientBlue);
+            if (ambientRed != 0.0f || ambientGreen != 0.0f || ambientBlue != 0.0f) {
```

Comment:
This  "if" test has caused a regression in behavior. To reproduce, run the `LightingSample`, show a Sphere, select one of the ambient lights, and then deselect it. Unchecking it should go back to black but no longer does.